R/lazyframe-frame.R

```diff
@@ -358,12 +759,2052 @@ lazyframe__slice <- function(offset, length = NULL) {
+  })
+}
+
+#' Rename column names
+#'
+#' @param mapping Either a function that takes a character vector as input and
+#' returns one as input, or a named list where names are old column names and
+#' values are the new ones.
+#' @param ... <[`dynamic-dots`][rlang::dyn-dots]> If `mapping` is missing,
+#' those values are used.
+#' @param strict Validate that all column names exist in the current schema,
+#' and throw an error if any do not. (Note that this parameter is a no-op when
+#' passing a function to `mapping`).
+#'
+#' @details
+#' If existing names are swapped (e.g. 'A' points to 'B' and 'B' points to
+#' 'A'), polars will block projection and predicate pushdowns at this node.
+#'
+#' @inherit as_polars_lf return
+#' @examples
+#' lf <- pl$LazyFrame(
+#'   foo = 1:3,
+#'   bar = 6:8,
+#'   ham = letters[1:3]
+#' )
+#'
+#' lf$rename(foo = "apple")$collect()
+#'
+#' lf$rename(
+#'   \(column_name) paste0("c", substr(column_name, 2, 100))
+#' )$collect()
+lazyframe__rename <- function(mapping, ..., strict = TRUE) {
+  wrap({
+    if (!missing(mapping) && is_function(mapping)) {
+      check_dots_empty0(...)
+      self$select(pl$all()$name$map(mapping))
+    } else {
+      if (missing(mapping) || !is.list(mapping)) {
+        mapping <- list2(...)
+      }
+      existing <- names(mapping)
+      new <- unlist(mapping)
+      self$`_ldf`$rename(existing, new, strict)
+    }
+  })
+}
+
+#' Fetch `n` rows of a LazyFrame
+#'
+#' This is similar to `$collect()` but limit the number of rows to collect. It
+#' is mostly useful to check that a query works as expected.
+#'
+#'
+#' @details
+#' `$fetch()` does not guarantee the final number of rows in the DataFrame output.
+#' It only guarantees that `n` rows are used at the beginning of the query.
+#' Filters, join operations and a lower number of rows available in the scanned
+#' file influence the final number of rows.
+#'
+#' @param n_rows Integer. Maximum number of rows to fetch.
+#' @inheritParams lazyframe__collect
+#' @return A DataFrame of maximum n_rows
+#' @seealso
+#'  - [`$collect()`][lazyframe__collect] - regular collect.
+#'  - [`$profile()`][lazyframe__profile] - same as `$collect()` but also returns
+#'    a table with each operation profiled.
+#'  - [`$collect_in_background()`][lazyframe__collect_in_background] - non-blocking
+#'    collect returns a future handle. Can also just be used via
+#'    `$collect(collect_in_background = TRUE)`.
+#'  - [`$sink_parquet()`][lazyframe__sink_parquet()] streams query to a parquet file.
+#'  - [`$sink_ipc()`][lazyframe__sink_ipc()] streams query to a arrow file.
+#'
+#' @examples
+#' # fetch 3 rows
+#' as_polars_lf(iris)$fetch(3)
+#'
+#' # this fetch-query returns 4 rows, because we started with 3 and appended one
+#' # row in the query (see section 'Details')
+#' as_polars_lf(iris)$
+#'   select(pl$col("Species")$append("flora gigantica, alien"))$
+#'   fetch(3)
+lazyframe__fetch <- function(
```

Comment:
```suggestion
lazyframe___fetch <- function(
```

Reply:
I removed this function, per your comment above